spyder/preferences/languageserver.py

```diff
@@ -203,6 +208,12 @@ def __init__(self, parent, language=None, cmd='', host='127.0.0.1',
         self.external_cb.setChecked(external)
         self.external_cb.stateChanged.connect(self.set_local_options)
 
+        self.stdio_cb = QCheckBox(_('Use stdio pipes for communication'), self)
```

Comment:
This is not correct. Since using an external server and stdio are mutually exclusive options, you need to use radio buttons for both here with the same button group.

Please see how we do that for the screen resolution options here:

https://github.com/spyder-ide/spyder/blob/ffb1f9361c0f6357b13f2f45b6dfe4f9e17cea70/spyder/preferences/general.py#L192-L234

Reply:
But we may have TCP "internal" servers that are launched by Spyder

Reply:
I think I expressed myself in a wrong way. stdio implies not tcp

Reply:
Sorry, where is this option being added? In the "Other languages" dialog you designed? Or in the "Advanced settings" we added with Daniel?

Reply:
On both panels, to be consistent

Reply:
Ok, then I think for "Advanced settings" we should really use ratio buttons for "External server" and "Stdio" because both can't be selected at the same time, right?

Reply:
But, both can be unselected at the same time, i.e., An internal TCP server

Reply:
You're right. But happens when users try to select both at the same time?

Reply:
They simply can't. Once an option is selected, the other is unchecked and disabled

Reply:
Ok, great!